ERCS/erc-7902.md

```diff
@@ -0,0 +1,206 @@
+---
+eip: 7902
+title: Wallet Capabilities for Account Abstraction
+description: EIP-5792 Capabilities allowing dApps and wallets to exchange all AA specific UserOp fields
+author: Yoav Weiss (@yoavw), Alex Forshtat (@forshtat), Dror Tirosh (@drortirosh), Shahaf Nacson (@shahafn)
+discussions-to: https://ethereum-magicians.org/t/erc-wallet-capabilities-for-account-abstraction/23122
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-03-01
+requires: 5792, 7702
+---
+
+## Abstract
+
+[EIP-5792](./eip-5792) defines a baseline JSON-RPC API for a communication between wallets and dapps,
+and provides an ability to extend the base protocol with "capabilities".
+
+This proposal defines a set of "capabilities" the wallets may want to implement in order to provide a comprehensive
+support for Account Abstraction (AA).
+
+These "capabilities" enable passing any data that may be required when using a Paymaster contract,
+allows limiting the time range during which the UserOperation is considered valid,
+provides a way for the dApp to manually control the semi-abstracted nonce and AA-specific gas limits of the UserOp,
+and even allow the dApp to take part in selecting the [EIP-7702](./eip-7702) account implementation.
+
+## Motivation
+
+[ERC-4337](./eip-4337.md) introduced Account Abstraction, enabling Smart Contract Accounts to function as first-class citizens in Ethereum.
+However, while [ERC-4337](./eip-4337.md) and [ERC-7769](./eip-7769.md) define a low-level RPC API for Account Abstraction,
+they do not specify a way for advanced AA-aware dApps to communicate their supported features and parameters to the advanced AA Wallet Applications.
+
+This ERC addresses the issue by defining a structured set of capabilities tailored for AA-aware dApps and Wallet Applications.
+
+It utilises the [EIP-5792](./eip-5792) wallet capability model to express some of the critical aspects of AA,
+ensuring dApps can seamlessly adapt to different AA Wallets without requiring custom solutions.
+
+## Specification
+
+All actions in Account Abstraction within the context of EIP-5792 must be done on a single chain and atomically.
+
+We define the following list of new "capabilities" which together cover many features necessary for Account Abstraction.
+Note that use of Paymasters managed by a "paymaster web service" is described in [ERC-7677](./eip-7677).
+
+### Create [EIP-7702](./eip-7702) Authorization Capability
+
+This capability is designed to be used with [EIP-7702](./eip-7702) and requests the Wallet Application to provide
```

Comment:
not sure what is the use-case for application to require specific account implementation from the wallet.
- the wallet MUST NOT accept arbitrary delegations - it must explicitly assert the delegate is known to the wallet, it supports the basic usage (e.g. "executeBatch"), and doesn't contain malicious code.
- If the application requires a specific feature from the wallet - e.g. session-keys - it can't simply ask for a new delegate: there must be a separate API (e.g. an explicit sendCalls capability)

Reply:
that is: if an app wants a specific feature (normal batching is part of 5792, but say it wants something else, like session keys) - then there should be a specific capability for that, not a "low-level" setting of the auth field.